server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -248,4 +262,23 @@ private PropertiesConfiguration getConfiguration(PropertiesProvider propertiesPr
             return propertiesConfiguration;
         }
     }
+
+    public static class CassandraSchemaChecker implements ConfigurationPerformer {
+        private final CassandraSchemaVersionManager versionManager;
+
+        @Inject
+        public CassandraSchemaChecker(CassandraSchemaVersionManager versionManager) {
+            this.versionManager = versionManager;
+        }
+
+        @Override
+        public void initModule() {
+            versionManager.ensureSchemaIsSupported();
```

Comment:
you should take the business decision about what to do with the schema version here.